Lesson_5/Backpack.java

```diff
@@ -0,0 +1,59 @@
+package Algorithms_and_Data_Structure.Homework.Lesson_5;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Backpack {
+
+    int totalWeight;
+    Set<List<Item>> possibleSets = new LinkedHashSet<>();
+    int bestValue;
+
+
+    public Backpack(int totalWeight) {
+        this.totalWeight = totalWeight;
+    }
+
+    public static void main(String[] args) {
+        List<Item> items = new ArrayList<>();
+        items.add(new Item(1, 700));
+        items.add(new Item(3, 1000));
+        items.add(new Item(4, 2000));
+        items.add(new Item(2, 600));
+        items.add(new Item(1, 500));
+
+
+        Backpack backpack = new Backpack(6);
+
+        System.out.println(backpack.getBestValue(items));
+    }
+
+    private int getBestValue(List<Item> list) {
+        bestValue(list.size(), list);
+        for (List<Item> l: possibleSets) {
+            int sumWeight = 0, sumValue = 0;
+            for (Item i: l) {
```

Comment:
однобуквенные именования - не самый лучший подход. 
```suggestion
            for (Item item: items) {
```